### PR TITLE
Log asserts

### PR DESCRIPTION
**Pull request by sbc100**
Log assert messages to the console.

Also, rename/cleanup CCScriptSupport::executeAssert.

### CODE REVIEW

File: cocos2dx/script_support/CCScriptSupport.h [hunk 1]
```diff
@@ -225,8 +225,11 @@ class CC_DLL CCScriptEngineProtocol
 
     /** function for common event */
     virtual int executeEvent(int nHandler, const char* pEventName, CCObject* pEventSource = NULL, const char* pEventSourceClassName = NULL) = 0;
-    /** function for assert test */
-    virtual bool executeAssert(bool cond, const char *msg = NULL) = 0;
+
+    /** called by CCAssert to allow scripting engine to handle failed assertions
+     * @return true if the assert was handled by the script engine, false otherwise.
+     */
+    virtual bool handleAssert(const char *msg) = 0;
```
Comment (dumganhar) [inline cocos2dx/script_support/CCScriptSupport.h:232]: Yes, I like `handleAssert` more than `executeAssert`, but do you think other `executeXXX` also need to be renamed to `handleXXX`?
For instance, `executeEvent` ~> `handleEvent` ?

Comment (sbc100) [inline cocos2dx/script_support/CCScriptSupport.h:232]: Maybe.  I'm don't feel strongly about the others.

I changed this one as it isn't really executing a callback, its deciding if the script engine should handle/display
the assert.   At least that is my understanding of the purpose of this code.   It doesn't call use code/handler at all.

**Comment by sbc100** [comment]
Rebased onto master.  Please take another look.

File: cocos2dx/include/ccMacros.h [hunk 2]
```diff
@@ -36,12 +36,14 @@ of this software and associated documentation files (the "Software"), to deal
...
-#define CCAssert(cond, msg)                             \
-    {                                                   \
-        cc_assert_script_compatible(!!(cond), (msg));   \
-        CC_ASSERT(cond);                                \
-    }
+extern bool CC_DLL cc_assert_script_compatible(const char *msg);
+#define CCAssert(cond, msg) do {                              \
+      if (!(cond)) {                                          \
+        if (!cc_assert_script_compatible(msg) && strlen(msg)) \
+          cocos2d::CCLog("Assert failed: %s", msg);           \
```
Comment (dumganhar) [inline cocos2dx/include/ccMacros.h:43]: Is it better to output filename and line number for cocos2d::CCLog?

Comment (sbc100) [inline cocos2dx/include/ccMacros.h:43]: The actual call to assert below should include the line number.  It was only the condition which was missing.. At least on linux and nacl.

Comment (dumganhar) [inline cocos2dx/include/ccMacros.h:43]: That makes sense.

**Comment by dumganhar** [comment]
Thank you.

**Pull request merged**